redfish-core/lib/assembly.hpp

```diff
@@ -176,6 +176,45 @@ inline void
                             "LocationCode",
                             "LocationCode");
                     }
+                    else if (interface == "xyz.openbmc_project.Inventory.Item")
+                    {
+                        crow::connections::systemBus->async_method_call(
+                            [aResp, assemblyIndex](
+                                const boost::system::error_code ec2,
+                                const std::variant<bool>& property) {
+                            if (ec2)
+                            {
+                                BMCWEB_LOG_DEBUG << "DBUS response error";
+                                messages::internalError(aResp->res);
+                                return;
+                            }
+
+                            nlohmann::json& assemblyArray =
+                                aResp->res.jsonValue["Assemblies"];
+                            nlohmann::json& assemblyData =
+                                assemblyArray.at(assemblyIndex);
+
+                            const bool* value = std::get_if<bool>(&property);
+
+                            if (value == nullptr)
+                            {
+                                // illegal value
+                                messages::internalError(aResp->res);
+                                return;
+                            }
+                            if (!*value)
+                            {
+                                assemblyData["Status"]["State"] = "Absent";
+                            }
+                            else
+                            {
+                                assemblyData["Status"]["State"] = "Enabled";
+                            }
+                            },
+                            serviceName, assembly,
+                            "org.freedesktop.DBus.Properties", "Get",
```

Comment:
All upstream code is using `sdbusplus::asio::getProperty<bool>(`
https://github.com/openbmc/bmcweb/blob/3ccb3adb9a14783f6bef601506de9f8bcae22d51/redfish-core/lib/led.hpp#L61